packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -1526,3 +1533,21 @@ function buildParameterMap(parameters: {
 
   return parameterMap;
 }
+
+/**
+ * Remove any template elements that we don't want to show users.
+ */
+function obscureTemplate(template: any = {}) {
+  if (template.Rules) {
+    // see https://github.com/aws/aws-cdk/issues/17942
+    if (template.Rules.CheckBootstrapVersion) {
+      if (Object.keys(template.Rules).length > 1) {
+        delete template.Rules.CheckBootstrapVersion;
```

Comment:
I thought you had mentioned that the rules would only be deleted if the check bootstrap version was the only rules. Did I misinterpret that, or did we switch to always removing the bootstrap version check?

Reply:
The intention is to always remove the Rule, but to remove the entire Rules section if that is the only rule.